### PR TITLE
MaybePrefetchOverflowPages

### PR DESCRIPTION
**Pull request by aviviadi**

### CODE REVIEW

File: src/Voron/Impl/Paging/AbstractPager.cs [hunk 1]
```diff
@@ -258,5 +261,28 @@ public static bool IsKeySizeValid(int keySize)
 
             return true;
         }
+
+        public void MaybePrefetchMemory(List<long> pagesToPrefetch)
```
Comment (ayende) [inline src/Voron/Impl/Paging/AbstractPager.cs:265]: - We need to move this to Win32 pager.
- We need to implement this via madvise(WILL_NEED) in Linux

File: src/Voron/Data/BTrees/TreeIterator.cs [hunk 2]
```diff
@@ -132,6 +137,14 @@ public bool MovePrev()
             return false;
         }
 
+        private void MaybePrefetchOverflowPages(TreePage page)
+        {
+            if (Sparrow.Platform.Platform.CanPrefetch)
```
Comment (redknightlois) [inline src/Voron/Data/BTrees/TreeIterator.cs:142]: Should be `Sparrow.Platform.CanPrefetch` instead.

File: src/Voron/Data/BTrees/TreeIterator.cs [hunk 3]
```diff
@@ -20,10 +21,11 @@ public unsafe class TreeIterator : IIterator
         private Slice _currentKey = default(Slice);
         private Slice _currentInternalKey = new Slice();
 
-        public TreeIterator(Tree tree, LowLevelTransaction tx)
+        public TreeIterator(Tree tree, LowLevelTransaction tx, bool prefetch)
         {
```
Comment (redknightlois) [inline src/Voron/Data/BTrees/TreeIterator.cs:25]: You can use:
`public TreeIterator(Tree tree, LowLevelTransaction tx, bool prefetch = false )`
and that will requiere less changes to the code and also make it read better. Will only have a parameter there when we explicitely want to control the prefetching.

Comment (aviviadi) [inline src/Voron/Data/BTrees/TreeIterator.cs:25]: I remember Oren asked to leave the signature like this, for explicitly purpose.

**Pull request merged**